packages/devtools_app/lib/src/flutter/custom_pointer_scroll_view.dart

```diff
@@ -0,0 +1,831 @@
+      // Only express interest in the event if it would actually result in a scroll.
+      if (targetScrollOffset != position.pixels) {
+        GestureBinding.instance.pointerSignalResolver
+            .register(event, _handlePointerScroll);
+      }
+    }
+  }
+
+  void _handlePointerScroll(PointerEvent event) {
+    assert(event is PointerScrollEvent);
+    if (_physics != null && !_physics.shouldAcceptUserOffset(position)) {
+      return;
+    }
+    final double targetScrollOffset =
+        _targetScrollOffsetForPointerScroll(event as PointerScrollEvent);
+    if (targetScrollOffset != position.pixels) {
+      position.jumpTo(targetScrollOffset);
+    }
+  }
+
+  // DESCRIPTION
+
+  @override
+  Widget build(BuildContext context) {
+    assert(position != null);
+    // _ScrollableScope must be placed above the BuildContext returned by
+    // notificationContext so that we can get this _CustomPointerScrollableState
+    // by doing the following:
+    //
+    // ScrollNotification notification;
+    // CustomPointerScrollable.of(notification.context)
+    //
+    // Since notificationContext is pointing to _gestureDetectorKey.context, _ScrollableScope
+    // must be placed above the widget using it: RawGestureDetector
+    Widget result = _ScrollableScope(
+      scrollable: this,
+      position: position,
+      // TODO(ianh): Having all these global keys is sad.
+      child: Listener(
+        behavior: HitTestBehavior.opaque,
+        onPointerSignal:
+            widget.customPointerSignalHandler ?? _receivedPointerSignal,
+        child: RawGestureDetector(
+          key: _gestureDetectorKey,
+          gestures: _gestureRecognizers,
+          excludeFromSemantics: widget.excludeFromSemantics,
+          child: Semantics(
+            explicitChildNodes: !widget.excludeFromSemantics,
+            child: IgnorePointer(
+              key: _ignorePointerKey,
+              ignoring: _shouldIgnorePointer,
+              ignoringSemantics: false,
+              child: widget.viewportBuilder(context, position),
+            ),
+          ),
+        ),
+      ),
+    );
+
+    if (!widget.excludeFromSemantics) {
+      result = _ScrollSemantics(
+        key: _scrollSemanticsKey,
+        child: result,
+        position: position,
+        allowImplicitScrolling: widget?.physics?.allowImplicitScrolling ??
+            _physics.allowImplicitScrolling,
+        semanticChildCount: widget.semanticChildCount,
+      );
+    }
+
+    return _configuration.buildViewportChrome(
+        context, result, widget.axisDirection);
+  }
+
+  @override
+  void debugFillProperties(DiagnosticPropertiesBuilder properties) {
+    super.debugFillProperties(properties);
+    properties.add(DiagnosticsProperty<ScrollPosition>('position', position));
+  }
+}
+
+// The following classes were copied from the Flutter framework, with the
+// exception that some types had to be replaced to use these classes with
```

Comment:
maybe replace: "that some types had to be replaced to use these classes with [CustomPointerScrollable]"
`with minor changes to use with [CustomPointerScrollable] instead of [???]`
that some types were replaced doesn't mean much to me and in the future more changes might be made so this comment could become misleading.